park-dashboard/src/Components/WeatherForecast/WeatherForecast.js

```diff
@@ -0,0 +1,311 @@
+import { BsFillCloudSlashFill, BsCloudsFill } from "react-icons/bs";
+import { GiRaining } from "react-icons/gi";
+import { Line } from "react-chartjs-2";
+import ChartDataLabels from "chartjs-plugin-datalabels";
+import Chart from "chart.js/auto";
+
+function WeatherForecast(item) {
+  const [dailyForecast, setDailyForecast] = useState([]);
+  const [latitude, setLatitude] = useState(null);
+  const [longitude, setLongitude] = useState(null);
+  const weatherForecastRef = useRef(null);
+  const parkCode = item.parkCode;
+  useEffect(() => {
+    const fetchWeatherData = async () => {
+      try {
+        const API_KEY = "92dbeb6cb1924c3a12889184257942ea";
+        const UNITS = "metric";
+        const response = await fetch(
+          `https://api.openweathermap.org/data/2.5/forecast?lat=${latitude}&lon=${longitude}&units=${UNITS}&appid=${API_KEY}`
+        );
+        const data = await response.json();
+        const dataList = data.list;
+        const groupedData = groupDataByDay(dataList);
+        const dailyData = calculateDailyData(groupedData);
+        setDailyForecast(dailyData);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+    if (latitude && longitude) {
+      fetchWeatherData();
+    }
+  }, [latitude, longitude]);
+
+  useEffect(() => {
+    const fetchParkData = async () => {
+      try {
+        const response = await fetch(
+          `https://developer.nps.gov/api/v1/parks?parkCode=${parkCode}&api_key=OaR7jmqSa22JAcsym9lVfStp58LmCqH9JdZUPEH7`
+        );
+        const data = await response.json();
+        const parkData = data.data;
+
+        if (parkData.length > 0) {
+          setLatitude(parkData[0].latitude);
+          setLongitude(parkData[0].longitude);
+        }
+      } catch (error) {
+        console.error("Error fetching park data:", error);
+      }
+    };
+    fetchParkData();
+  }, []);
+  useEffect(() => {
+    if (weatherForecastRef.current) {
+      const containerWidth = weatherForecastRef.current.offsetWidth;
+      const totalWidth = dailyForecast.length * 200;
+      if (totalWidth > containerWidth) {
+        weatherForecastRef.current.classList.add("scrollable");
+      } else {
+        weatherForecastRef.current.classList.remove("scrollable");
+      }
+    }
+  }, [dailyForecast]);
+
+  const groupDataByDay = (data) => {
+    const groupedData = {};
+    data.forEach((item) => {
+      const date = moment(item.dt_txt).format("YYYY-MM-DD");
+      if (!groupedData[date]) {
+        groupedData[date] = {
+          minTemperature: [],
+          maxTemperature: [],
+          description: item.weather[0].description,
+          icon: item.weather[0].icon,
+        };
+      }
+      groupedData[date].minTemperature.push(item.main.temp_min);
+      groupedData[date].maxTemperature.push(item.main.temp_max);
+    });
+    return groupedData;
+  };
+
+  const calculateDailyData = (groupedData) => {
+    const dailyData = [];
+    for (const date in groupedData) {
```

Comment:
I strongly suggest use of Object.entries on grouped data and then using map function for key values. Instead of manually creating an empty array and pushing values to it these ES6 functions might help to ease the functioning as a better practice. Although the code is working correctly and fulfilling the purpose.

Reply:
Thank you for your feedback, I have added the Object.entries on grouped data now.